dcicutils/ecr_utils.py

```diff
@@ -0,0 +1,68 @@
+import boto3
+import base64
+from .misc_utils import PRINT
+
+
+# Defines the standard Docker image tags for CGAP ECR
+# Latest and stable - something like this should be reasonable?
+CGAP_ECR_LAYOUT = {
+    'latest-wsgi': 'Latest version of WSGI Application',
+    'latest-indexer': 'Latest version of Indexer Application',
+    'latest-ingester': 'Latest version of the Ingester Application',
+    'stable-wsgi': 'Stable version of WSGI Application',
+    'stable-indexer': 'Stable version of Indexer Application',
+    'stable-ingester': 'Stable version of the Ingester Application'
+}
+
+
+class ECRUtils(object):
+    """ Utility class for interacting with ECR.
+        Initialized with an env name, from which a repo URL is resolved.
+
+        NOTE 1: unlike s3Utils, the resolved resources (repository) will NOT be
+        created if it does not already exist.
+        NOTE 2: the (already created) ECR repository must have the env_name in the URI
+    """
+
+    def __init__(self, *, env_name='cgap-mastertest', local_repository='cgap-wsgi'):
+        """ Creates an ECR client on startup """
+        self.env = env_name
+        self.local_repository = local_repository
+        self.is_cgap = 'cgap' in env_name
+        self.client = boto3.client('ecr', region='us-east-1')  # XXX: constant?
+        self.url = None  # set by calling the below method
+
+    def resolve_repository_uri(self, url=None):
+        if not self.url or url:
+            PRINT('NOTE: Calling out to ECR')
+            try:
+                resp = self.client.describe_repositories()
+                for repo in resp.get('repositories', []):
+                    if self.env in repo['repositoryUri']:
+                        return repo['repositoryUri']
+            except Exception as e:
+                PRINT('Could not retrieve repository information from ECR: %s' % e)
+        self.url = url  # hang onto this
+        return url
+
+    def authorize_user(self):
+        """ Calls to boto3 to get authorization credentials for ECR.
+            Gives a result like this:
+                {
+                    'authorizationData': [{'authorizationToken': 'very long token....',
+                                            'expiresAt': <datetime>,
+                                            'proxyEndpoint': <ecr URL> ... }]
+                }
+        """
+        try:
+            return self.client.get_authorization_token()['authorizationData'][0]
```

Comment:
I think this should be written in the other style that will catch the case of more than one bit of data returned, which would mean the query was ambiguous. e.g., (untested):
```
[auth_datum] = self.client.get_authorization_token()['authorizationData']
return auth_datum
```
or something like that.